### ISSUE TITLE
Bug: [prefer-optional-chain] don't report on `typeof` checks for globals

### ISSUE THREAD

**Issue by kaisermann** [typescript-eslint/typescript-eslint#7478]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.1.6&fileType=.tsx&code=GYVwdgxgLglg9mABBAFgUwgawJIGcBCATnAO65qEAUAlIgN4BQiihaUIhSlTziUAngAc0cYIhIwwAE1KIAhAF4FiAOTgpaYJLRSViAGT7xkmSQB0MiCAC2aMFAD8ZiKwCGUNAFEANmlv35ZTAQb28eagYAXyA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WloHsalfkwCG8WmQAWo5uii9o-aJHBgAviHVA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eFYDAruuGAL4g9A&tokens=true

### Repro Code

```typescript
function checkIsBrowser() {
  return (
    typeof window !== 'undefined' && window.document?.createElement != null
  )
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/prefer-optional-chain": "warn"
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strict": true
  }
}
```

### Expected Result

Keeps the function like the code example provided:

```js
function checkIsBrowser() {
  return (
    typeof window !== 'undefined' && window?.document?.createElement != null
  )
}
```

### Actual Result

It rewrites the condition to use the optional chain operator instead of the `typeof` comparison. The "fixed" code throws a `ReferenceError` if running in an environment where `window` is not defined.

```js
function checkIsBrowser() {
  return window?.document?.createElement != null
}
// Uncaught ReferenceError: window is not defined
```
